packages/expo-updates/android/src/main/java/expo/modules/updates/selectionpolicy/LoaderSelectionPolicyOverrideAware.kt

```diff
@@ -0,0 +1,55 @@
+package expo.modules.updates.selectionpolicy
+
+import expo.modules.updates.db.entity.UpdateEntity
+import expo.modules.updates.loader.UpdateDirective
+import org.json.JSONObject
+
+/**
+ * A LoaderSelectionPolicy that forces loading of updates when UpdatesConfigurationOverride is active.
+ * This enables one-time updates from different URLs/headers set via `setUpdateURLAndRequestHeadersOverride`.
+ * The policy always loads new updates regardless of commitTime.
+ */
+internal class LoaderSelectionPolicyOverrideAware : LoaderSelectionPolicy {
+  override fun shouldLoadNewUpdate(
+    newUpdate: UpdateEntity?,
+    launchedUpdate: UpdateEntity?,
+    filters: JSONObject?
+  ): Boolean {
+    if (newUpdate == null) {
+      return false
+    }
+    // if the new update doesn't pass its own manifest filters, we shouldn't load it
+    if (!SelectionPolicies.matchesFilters(newUpdate, filters)) {
+      return false
+    }
+    if (launchedUpdate == null) {
+      return true
+    }
+    return newUpdate.id != launchedUpdate.id
+  }
+
+  override fun shouldLoadRollBackToEmbeddedDirective(
```

Comment:
is the embedded update considered an override? i.e. if the URL is overridden and the update it gets is a roll-back-to-embedded directive, what's the behavior?

Reply:
i had the line below `return launchedUpdate.id == embeddedUpdate.id`. it happens when no updates available in the updateUrl, then we should still launch the embedded update.